src/common/path/get-absolute-path.injectable.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import path from "path";
+
+export type GetAbsolutePath = (...args: string[]) => string;
+
+const getAbsolutePathInjectable = getInjectable({
+  id: "get-absolute-path",
+
+  instantiate: (): GetAbsolutePath => path.resolve,
+
+  // This causes side effect e.g. Windows creates different kinds of
+  // absolute paths than linux
+  causesSideEffects: true,
```

Comment:
I disagree that this sort of thing should be overridden in tests, instead the tests should be written in an agnositc fasion.

Reply:
Unit tests shouldn’t have explicit side effects. All OS specific business logic should be unit tested without relying side-effects (here OS). 

Integration tests are something different, and those shouldn’t override this type of stuff. 

(keep in mind that current integration tests are not really integration tests, those are e2e smoke tests.)

1. It is enough to run unit tests with one operating system

2. Integration tests should be ran with all known operating systems

3. E2E tests should be ran with all known operating systems.

Reply:
I don't see how having this overriden doesn't diminish our trust in the unit test.

Reply:
A side-effect has no place in a unit test. As background, this particular side-effect already caused damage by only revealing its nasty flakiness once down in CI.

If trust in modules with side effects is diminished, one can write integration tests for them. This case however does not warrant such a thing, as there is no reasonable doubt that `path.resolve` would betray us.

Reply:
If there is no reasonable doubt that it would betray us then isn't unit test just wrong?

Reply:
If there is no reasonable doubt of operation for a minimal module with side-effects, we are free and right to stub it in unit testing.